accepted/0000-update-package-json.md

```diff
@@ -0,0 +1,61 @@
+# Change how `npm update` edits `package.json`
+
+## Summary
+
+`package.json` should not be edit by `npm update` unneccesarily. When updating a package to a newer version that still satisfies the original range, package.json should not change. Only when updating to a range that does not satisfy the range specified in `package.json` should it be changed and saved to `package.json`.
+
+## Motivation
+
+It's not expected that the ranges in `package.json` changes when running `npm update`. This was more clearly shown in `npm@5` when `--save` became the default behaviour. Previously, running `npm update` would not edit `package.json`. This unexpectedness also manifests itself in an especially bad way when updating packages that are specified with a non-standard range. For example, packages using tilde (~) when the project default is caret (^), since this will save the new range with the default.
```

Comment:
I don’t agree - when i run `npm update` i assume, and want, the lower range to be lifted.

Reply:
because it establishes a new minimum, and forces deduping in consumer trees, which pushes updates through the graph faster.

Reply:
Hm, I guess that the better behaviour then would be my other considered option 2:
> Make sure that npm keeps the range when updating. For example, updating `~1.0.0` when `1.0.1` is available will yield `~1.0.1`.

It solves my main gripe (non-default ranges not really being very usable) and would still keep the current behaviour of helping consumers get updates for nested dependencies.

Reply:
That part seems fine to me :-)

Reply:
https://github.com/npm/npm/issues/16813 is also related (I guess).

Reply:
@DanielRuf yup, as mentioned in the document! 😃 

@ljharb Considering your comments I will be rewriting this RFC to instead suggest that NPM should keep the ranges instead of always writing with the default. Seems like that would be a better solution, all in all.